File: 250111/체크판위에서 2/on-the-checkboard-2.cpp
#include <iostream>
using namespace std;

int main() {
    // Please write your code here.
    int R, C; char arr[15][15];
    cin >> R >> C;
    for (int i=0; i<R; i++) {
        for (int j=0; j<C; j++) {
            cin >> arr[i][j];
        }
    }

    int cnt = 0;
    char c = arr[0][0];
    for (int i=1; i<R-1; i++) {
        for (int j=1; j<C-1; j++) {
            if (arr[i][j] != c) {
                char arrIJ = arr[i][j];
                for (int k=i+1; k<R-1; k++) {
                    for (int m=j+1; m<C-1; m++) {
                        if (arr[k][m] != arrIJ) {
                            char arrKM = arr[k][m];
                            if (arrKM != arr[R-1][C-1]) cnt++;
                        }
                    }
                }
            }
        }
    }
    cout << cnt; 
    return 0;
}

File: 250629/체크판위에서 2/on-the-checkboard-2.py
R, C = map(int, input().split())
arr = [input().split() for i in range(R)]

cnt = 0
for i in range(1, R-1):
    for j in range(1, C-1):
        for k in range(i+1, R-1):
            for l in range(j+1, C-1):
                if arr[i][j] == arr[0][0]:
                    continue
                if arr[k][l] != arr[0][0]:
                    continue
                if arr[k][l] == arr[R-1][C-1]:
                    continue
                
                cnt += 1

print(cnt)